src/main/java/org/kohsuke/github/GHGistFile.java

```diff
@@ -62,10 +69,27 @@ public String getLanguage() {
     /**
      * Content of this file.
      *
+     * Content is lazy downloaded and cached locally the first time this method is called.
+     * Function modified for <a href="https://github.com/hub4j/github-api/issues/1325">issue #1325</a>.
+     *
      * @return the content
      */
     public String getContent() {
-        return content;
+        if (this.content != null) {
+            return this.content;
+        }
+        this.content = "";
+        if (this.size > 0) {
+            try {
+                InputStream inputStream = this.owner.root().createRequest()
+                        .setRawUrlPath(raw_url)
+                        .fetchStream(Requester::copyInputStream);
+                this.content = IOUtils.toString(inputStream, Charset.defaultCharset());
+            } catch (IOException e) {
+                throw new GHException("Failed to retrieve the gist file content", e);
+            }
+        }
+        return this.content;
```

Comment:
The way you have it written, if the first call to `getContent()` throws, the next call will always return `""`.  Is that what you think should happen?  

```suggestion
        if (this.content == null) {
            if (this.size == 0) {
                this.content = "";
            } else {
                try {
                    InputStream inputStream = this.owner.root().createRequest()
                            .setRawUrlPath(raw_url)
                            .fetchStream(Requester::copyInputStream);
                    this.content = IOUtils.toString(inputStream, Charset.defaultCharset());
                } catch (IOException e) {
                    throw new GHException("Failed to retrieve the gist file content", e);
                }
            }
        }
        return this.content;
```

Reply:
Ohhh!! Nice catch! Thank you so much for the suggestions! I will make my revision right away